ambry-protocol/src/main/java/com/github/ambry/protocol/RequestDropper.java

```diff
@@ -64,7 +66,17 @@ public void run() {
   }
 
   public void shutdown() throws InterruptedException {
+    running.set(false);
     requestChannel.sendRequest(EmptyRequest.getInstance());
```

Comment:
Do we need to append empty request anymore?

Reply:
Yeah, we may need it. `requestChannel.getDroppedRequests()` is a blocking call. Due to that, if there are no requests (for ex: a server is up but not taking any requests yet), the execution won't come out of the loop and doesn't check the `running` status. Server would only forcefully shutdown after 5 mins (timeout) in this case.